.travis.yml

```diff
@@ -19,51 +19,57 @@ env:
     # configuration.
     ###########################################################################
 
-    # Check the matrix of:
+    # Check a subset of the matrix of:
     #   LLVM  : {2.9, 3.4}
     #   SOLVERS : {Z3, STP, STP:Z3, metaSMT}
     #   STP_VERSION   : {2.1.2, master}
     #   METASMT_DEFAULT : {btor, stp, z3}
     #   UCLIBC: {0, klee_uclibc_v1.0.0, klee_0_9_29} // Note ``0`` means disabled
-    # with Asserts enabled.
+    #   DISABLE_ASSERTIONS: {0, 1}
+    #   ENABLE_OPTIMIZED: {0, 1}
+    #   COVERAGE: {0, 1}
+    #   USE_TCMALLOC: {0, 1}
+    #   USE_CMAKE: {0, 1}
 
     # COVERAGE set indicated that coverage data should be uplaoded to the server, only ONE job should have COVERAGE set
 
     matrix:
-    # Check KLEE CMake build
+    # FIXME: Enable when we want to test LLVM3.5
+    #- LLVM_VERSION=3.5 SOLVERS=STP:Z3 STP_VERSION=2.1.2 KLEE_UCLIBC=klee_uclibc_v1.0.0 DISABLE_ASSERTIONS=0 ENABLE_OPTIMIZED=1
+    #- LLVM_VERSION=3.5 SOLVERS=STP:Z3 STP_VERSION=2.1.2 KLEE_UCLIBC=klee_uclibc_v1.0.0 DISABLE_ASSERTIONS=0 ENABLE_OPTIMIZED=1 USE_CMAKE=1
+
+    # TODO: Add ASan build
+    # TODO: Add coverage build
+    # TODO: Add Doxygen build
+
+    # Check KLEE CMake build in a few configurations
     - LLVM_VERSION=3.4 SOLVERS=STP:Z3 STP_VERSION=2.1.2 KLEE_UCLIBC=klee_uclibc_v1.0.0 DISABLE_ASSERTIONS=0 ENABLE_OPTIMIZED=1 COVERAGE=0 USE_TCMALLOC=1 USE_CMAKE=1
     - LLVM_VERSION=3.4 SOLVERS=metaSMT METASMT_DEFAULT=STP KLEE_UCLIBC=klee_uclibc_v1.0.0 DISABLE_ASSERTIONS=0 ENABLE_OPTIMIZED=1 COVERAGE=0 USE_TCMALLOC=1 USE_CMAKE=1
     - LLVM_VERSION=2.9 SOLVERS=STP:Z3 STP_VERSION=2.1.2 KLEE_UCLIBC=klee_uclibc_v1.0.0 DISABLE_ASSERTIONS=0 ENABLE_OPTIMIZED=1 COVERAGE=0 USE_TCMALLOC=1 USE_CMAKE=1
```

Comment:
Unless their is a concrete timeline for removing LLVM 2.9 both the old build system I want to keep this. At the moment it might be the case that we deprecate the old build system before we deprecate LLVM 2.9 so I don't want the CMake build when using LLVM 2.9 to become broken.